Sources/MessagingPush/MessagingPush.swift

```diff
@@ -7,10 +7,10 @@ import Foundation
   */
 public class MessagingPush: ModuleTopLevelObject<MessagingPushInstance>, MessagingPushInstance {
     @Atomic public private(set) static var shared = MessagingPush()
-    @Atomic public private(set) static var moduleConfig: MessagingPushConfigOptions = .Factory.create()
     private static let moduleName = "MessagingPush"
 
     private var globalDataStore: GlobalDataStore
+    public var moduleConfig: MessagingPushConfigOptions { implementation?.moduleConfig ?? MessagingPushConfigBuilder().build() }
```

Comment:
Previously, we were just modifying the static config. Since it is updated with builder now, I thought it would be better to avoid static and reuse implementation instead.

Reply:
[Since the config can be injected into classes via digraph](https://github.com/customerio/customerio-ios/pull/560/files#diff-01f1b1edfc257ed559f8b272e971182e1daf2be00dc39a0c343bdbcc45dc5148R16-R20), what do you think about continuing to use the previous `static` implementation and update `MessagingPushImplementation(diGraph: diGraphShared, moduleConfig: config)` to `MessagingPushImplementation(diGraph: diGraphShared)`? 

The static solution is easier for me to read and confirm it's atomic compared to the solution in this PR.